logs/client.go

```diff
@@ -0,0 +1,200 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"math/rand/v2"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/solarwinds/swo-cli/version"
+)
+
+type ColorStrFunc func(format string, a ...interface{}) string
+
+var colors = []ColorStrFunc{
+	color.CyanString,
+	color.YellowString,
+	color.GreenString,
+	color.MagentaString,
+	color.RedString,
+}
+
+type Client struct {
+	opts       *Options
+	httpClient http.Client
+	output     *os.File
+}
+
+type Log struct {
+	Time     time.Time `json:"time"`
+	Message  string    `json:"message"`
+	Hostname string    `json:"hostname"`
+	Severity string    `json:"severity"`
+	Program  string    `json:"program"`
+}
+
+type PageInfo struct {
+	PrevPage string `json:"prevPage"`
+	NextPage string `json:"nextPage"`
+}
+
+type LogsData struct {
+	Logs     []Log `json:"logs"`
+	PageInfo `json:"pageInfo"`
+}
+
+func NewClient(opts *Options) (*Client, error) {
+	return &Client{
+		httpClient: *http.DefaultClient,
+		opts:       opts,
+		output:     os.Stdout,
+	}, nil
+}
+
+func (c *Client) prepareRequest(ctx context.Context) (*http.Request, error) {
+	logsEndpoint, err := url.JoinPath(c.opts.ApiUrl, "v1/logs")
+	if err != nil {
+		return nil, err
+	}
+
+	logsUrl, err := url.Parse(logsEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	params := url.Values{}
+	params.Add("pageSize", strconv.Itoa(c.opts.count))
+	if c.opts.group != "" {
+		params.Add("group", c.opts.group)
+	}
+	if c.opts.minTime != "" {
+		params.Add("startTime", c.opts.minTime)
+	}
+	if c.opts.maxTime != "" {
+		params.Add("endTime", c.opts.maxTime)
+	}
+
+	var filter string
+	if c.opts.system != "" {
+		filter = fmt.Sprintf("host:%s", c.opts.system)
+	}
```

Comment:
This is an interesting one. Papertrail's CLI would actually lookup the hostname, translate it to an ID, and use that. So it was looking for a substring match, but it would only match a single sender. So for example, if you pass `-s HOST` when there's both `HOSTA` and `HOSTB`, it'll just look for logs from HOSTA.

What you have here will just do a substring match, returning logs from both `HOSTA` and `HOSTB` in that case. Honestly that's probably less confusing behavior, but it is a change from Papertrail. (And the API doesn't support the exact same behavior.)

I think we should keep it this way, but just be aware that we might see some questions about it.

Reply:
changed to `filter = fmt.Sprintf(`host:"%s"`, c.opts.system)`